src/main/scala/codacy/pylint/Pylint.scala

```diff
@@ -10,50 +11,129 @@ import scala.util.{Properties, Success, Try}
 
 object Pylint extends Tool {
 
-  override def apply(path: Path, conf: Option[Seq[PatternDef]], files: Option[Set[Path]])(implicit spec: Spec): Try[Iterable[Result]] = {
-    def isEnabled(issue: Issue) =
-      conf.map(_.exists(_.patternId == issue.patternId)).getOrElse(true)
+   override def apply(path: Path, conf: Option[Seq[PatternDef]], files: Option[Set[Path]])(implicit spec: Spec): Try[Iterable[Result]] = {
+    def isEnabled(issue: Result) = {
+       issue match {
+          case Issue(_, _, patternId, _) => conf.map(item => item.exists(_.patternId == patternId)).getOrElse(true)
+          case _ => true
+       }
+    }
 
-    commandFor(path, conf, files).flatMap { case cmd =>
-      val wrappedCmd = Seq("bash", "-c", cmd.mkString(" "))
+    def buildFileCommands(files: Map[String, Array[String]]) = {
+      files.map { case (key, values) =>
+          commandFor(key, path, conf, values).map (item => Seq(item).mkString(" "))}
+                     .flatMap( item => item.toOption)
+    }
 
-      Try(wrappedCmd.lineStream_!(ProcessLogger(_ => ()))).map { lines =>
-        lines.flatMap(parseLine).filter(isEnabled)
-      }
+    val collectedFiles = collectFiles(files, path)
+    val classified = classifyFiles(collectedFiles)
+    val commands = classified.map { case item => buildFileCommands(item) }
+    val iterators = commands.map {
+        case item => item.map { cmd => cmd.lineStream_!(ProcessLogger(_ => ()))}
```

Comment:
We should not use the scala process directly. In the code you have access to our `CommandRunner` which wrapps this code.